lib/sonata/tip/getChannelTipInfo.ts

```diff
@@ -0,0 +1,31 @@
+import pregenerateChannelWallet from "../../privy/pregenerateChannelWallet";
+import searchChannels from "../../privy/searchChannels";
+
+const FEE = .1;
+
+function getChannelId(url: string) {
+  const match = url.match(/^(\w+):\/\/([^/]+)\/channel\/(\w+)$/);
+  if (match) return match[3];
+  return null;
+}
+
+async function getChannelTipInfo(referer = '', amount: number | null = null) {
+  const channelId = getChannelId(referer);
+  if (!channelId) return null;
+  try {
+    let [channel] = await searchChannels([channelId]);
+
+    if (!channel) {
+      const response = await pregenerateChannelWallet(channelId);
+      channel = await response.json();
+    }
+    const account = channel.linked_accounts.find(account => account.type === 'wallet')!;
+
+    return { channelAmount: Number(amount ?? 0) * FEE, channelAddress: account.address };
+  } catch (error) {
+    console.error(error)
+    return null;
+  }
+}
```

Comment:
can you help me understand the logic here?
- expected: only channels in out `official channel list` will receive channel tips. all other channels will `skip` channel tip with 100% of tip going to the curator.
- question: is this calling `pregenerate` for ANY channel? or, only those supported in our list of `17 official channels`?